symengine/lib/symengine_wrapper.pyx

```diff
@@ -508,6 +508,60 @@ cdef class Basic(object):
     def _symbolic_(self, ring):
         return ring(self._sage_())
 
+def series(ex, x=None, x0=0, n=6, method='sympy', removeO=False):
+    # TODO: check for x0 an infinity, see sympy/core/expr.py
+    # TODO: nonzero x0
+    # TODO: order term
+    # underscored local vars are of symengine.py type
+    cdef Basic _ex = sympify(ex)
+    cdef Symbol _x = sympify(x)
+    if method == 'sympy':
+        from sympy import series as sy_series
+        return sy_series(_ex._sympy_(), _x._sympy_(), x0, n)
+    elif method == 'ring_series':
+        from sympy.polys.ring_series import rs_series
+        return rs_series(_ex._sympy_(), _x._sympy_(), n).as_expr().subs(x,x-x0)
+    elif method != 'symengine':
+        raise ValueError('unknown method in series()')
+
+    syms = _ex.free_symbols
+    if not syms:
+        return _ex
+    if len(syms) > 1:
+        from sympy import series as sy_series
+        return sy_series(_ex._sympy_(), _x._sympy_(), x0, n)
+    if _x is None:
+        _x = list(syms)[0]
+
+    if not _x in syms:
+        return _ex
+
+    cdef RCP[const symengine.Symbol] X = symengine.rcp_static_cast_Symbol(_x.thisptr)
+    cdef unsigned int N = n
+    cdef umap_short_basic umap
+    cdef umap_short_basic_iterator iter, iterend
+    cdef Basic coef
+
+    try:
+        umap = symengine.series(_ex.thisptr, X, N)
+    except:
```

Comment:
What kind of exceptions is this catching? One shouldn't use general except clause like this, but rather catch specific exceptions. Is this testing if symengine throws the "not implemented" exception?